Vagrantfile

```diff
@@ -172,7 +171,23 @@ Vagrant.configure("2") do |config|
         end
       when 'public_network'
         options[:ip] = vm_network['ip'] unless vm_network['ip'].nil? or vm_network['ip'].strip().empty?
-        options[:bridge] = vm_network['bridge'] unless vm_network['bridge'].nil? or vm_network['bridge'].empty?
+        bridge_interface = ""
+        bridge_interface = vm_network['bridge'] unless vm_network['bridge'].nil? or vm_network['bridge'].empty?
+
+        bridge_interface = get_default_nic() if bridge_interface.empty? and vm_network['auto_bridge_default_network']
+        options[:bridge] = bridge_interface unless bridge_interface.empty?
+
+        if vm_network['resuse_mac_address']
+          if File.exist?(File.dirname(__FILE__) + '/.publicMACAddress')
```

Comment:
should use lower case file names, .public-mac-address instead and don't forget to add this to .gitignore